% --------------------------------------------------
% 
% This chapter is for one-page cv
% 
% --------------------------------------------------

\chapter{Short CV}
\label{cha:cv}


\section*{Education}

\begin{itemize}

	\item{
      \textbf{Doctor of Philosophy} (2018 -- 2022), 
      University of Crete, \href{https://www.biology.uoc.gr/en}{Biology Department} \\
      \textbf{Thesis:}
      Microbial communities through the lens of high throughput sequencing, 
      data integration and metabolic networks analysis \\
      Thesis conducted at \href{https://imbbc.hcmr.gr/}{IMBBC - HCMR}
   }

	\item{
      \textbf{M.Sc. in  Bioinformatics} (2016 -- 2018), 
      University of Crete, \href{http://www.english.med.uoc.gr/}{School of Medicine} \\ 
      \textbf{\href{https://www.openarchives.gr/aggregator-openarchives/edm/elocus/000018-dlib_5_2_f_metadata-dlib-1545038085-364284-26948.tkl}{Thesis}:}
      eDNA metabarcoding for biodiversity assessment: Algorithm design and bioinformatics analysis pipeline implementation \\
      Thesis conducted at \href{https://imbbc.hcmr.gr/}{IMBBC - HCMR}
   }
    
	\item{
      
      \textbf{B.Sc. in Biology} (2011 -- 2016), 
      National and Kapodistrian University of Athens, \href{http://en.biol.uoa.gr/}{department of Biology}}\\ 
      \textbf{Thesis:}
		Morphology, morphometry and anatomy of species of the genus \textit{Pseudamnicola} in Greece
		
\end{itemize}



% ------------------- PROJECTS -------------------

\section*{Research projects - working Experience}

\begin{itemize}

   \item {
      \textbf{A workflow for marine Genomic Observatories data analysis} (2021 - ongoing) \\
      \textbf{Role:} scientific responsiblse \& developer \\ 
      This \href{https://www.eosc-life.eu/opencall/}{EOSC-Life} funded project aims at 
      developing a workflow for the analysis of EMBRC's Genomic Observatories (GOs) data, allowing researchers to deal better with this increasing amount of the data and make them more easily interpretable.
   }


   \item{
      \textbf{PREGO: Process, environment, organism (PREGO)} (2019 - 2021) \\ 
      \textbf{Role:} PhD candidate \\
      \href{http://prego.hcmr.gr/}{PREGO} is a systems-biology approach to elucidate ecosystem function at the microbial dimension.
   }

   \item {
      \textbf{ELIXIR-GR} (2019 - 2021) \\ 
      \textbf{Role:} technical support \\
      {\href{https://elixir-greece.org}{ELIXIR-GR}} is the Greek National Node of the ESFRI \href{https://elixir-europe.org/}{European RI ELIXIR}, 
      a distributed e-Infrastructure aiming at the construction of a sustainable European infrastructure for biological information.}

   \item{
      \textbf{RECONNECT} (2018 - 2020) \\
      \textbf{Role:} technical support \\ 
      \href{https://reconnect.hcmr.gr/}{RECONNECT} is an Interreg V-B "Balkan-Mediterranean 2014-2020" project. 
      It aims to develop strategies for sustainable management of Marine Protected Areas (MPAs) and Natura 2000 sites.   
   }

\end{itemize}




% ------------------- PUBLICATIONS -------------------

\section*{Publications}

\begin{itemize}

   \item{
      \textbf{PREGO: A Literature and Data-Mining Resource to Associate Microorganisms, Biological Processes, and Environment Types.} \\ 
      \textbf{Zafeiropoulos, H.}, Paragkamian S.\footnote{ZH and PS contibuted equally in this study}, 
      Stelios Ninidakis, Georgios A. Pavlopoulos, Lars Juhl Jensen, and Evangelos Pafilis. \textit{Microorganisms} 10, no. 2 (2022): 293.,
      DOI: \href{https://doi.org/10.3390/microorganisms10020293}{10.3390/microorganisms10020293}
   }



   \item{
      \textbf{The Dark mAtteR iNvestigator (DARN) tool: getting to know the known unknowns in COI amplicon data} \\
      \textbf{Zafeiropoulos H.}, Gargan L., Hintikka S., Pavloudi C., \& Carlsson J. \textit{Metabarcoding and Metagenomics}, 5, p.e69657, 2021, 
      DOI: \href{https://doi.org/10.3897/mbmg.5.69657}{10.3897/mbmg.5.69657} 
   }

   \item{
      \textbf{0s \& 1s in marine molecular research: a regional HPC perspective.} \\
      \textbf{Zafeiropoulos H.}, Gioti A., Ninidakis S., Potirakis A., ..., \& Pafilis E. \textit{GigaScience},  9(3), p.giab053, 2021
      DOI: \href{https://doi.org/10.1093/gigascience/giab053}{10.1093/gigascience/giab053}
   }

   \item{
      \textbf{Geometric Algorithms for Sampling the Flux Space of Metabolic Networks} \\
      Chalkis, A., Fisikopoulos, V., Tsigaridas, E. \& \textbf{Zafeiropoulos, H.} \textit{37th International Symposium on Computational Geometry (SoCG 2021)}, 21:1--21:16, 189, 2021
      DOI: \href{https://drops.dagstuhl.de/opus/volltexte/2021/13820/}{10.4230/LIPIcs.SoCG.2021.21}
   }

   \item{
      \textbf{The Santorini Volcanic Complex as a Valuable Source of Enzymes for Bioenergy} \\ 
      Polymenakou, P.N., Nomikou, P., \textbf{Zafeiropoulos, H.}, Mandalakis, M., Anastasiou, T.I., Kilias, S., Kyrpides, N.C., Kotoulas, G. \& Magoulas,A. \textit{Energies}, 14(5), p.1414, 2021
      DOI: \href{https://doi.org/10.3390/en14051414}{10.3390/en14051414}
   }

   \item{
      \textbf{PEMA: a flexible Pipeline for Environmental DNA Metabarcoding Analysis of the 16S/18S ribosomal RNA, ITS, and COI marker genes} \\ 
      \textbf{Zafeiropoulos, H.}, Viet, H.Q., Vasileiadou, K., Potirakis, A., Arvanitidis, C., Topalis, P., Pavloudi, C. \& Pafilis, E. \textit{GigaScience}, 9(3), p.giaa022, 2020
      DOI: \href{https://doi.org/10.1093/gigascience/giaa022}{10.1093/gigascience/giaa022}
   }

\end{itemize}


\subsection*{In preparation}

\begin{itemize}
   \item \texttt{dingo}: a Python library for metabolic networks analysis 
   \item Deciphering the functional potential of a hypersaline swamp microbial mat community
\end{itemize}





% ------------------- AWARDS -------------------

\section*{Awards}

\begin{itemize}

   \item{
      \textbf{\href{https://www.embo.org/funding/fellowships-grants-and-career-support/scientific-exchange-grants/}{European Molecular Biology Organization Short-Term Fellowship}} (2022) \\ 
      \textbf{Project title:} Exploiting data integration, text-mining and computational geometry to enhance microbial interactions inference from  co-occurrence networks \\
      \href{Report}{https://hariszaf.github.io/microbetag/}

   }

   \item{
      \textbf{\href{https://www.mikrobiokosmos2021.org/}{Mikrobiokosmos travel grant in memorium of Prof. Kostas Drainas}} (2021) 
   }

   \item{
      \textbf{\href{https://summerofcode.withgoogle.com/}{Google Summer of Code}} (2021) \\
      \textbf{Project title:} From DNA sequences to metabolic interactions: building a pipeline to extract key metabolic processes \\
      \href{https://summerofcode.withgoogle.com/archive/2021/projects/6407348884602880}{Report}, \href{https://hariszaf.github.io/gsoc2021/}{GSoC archive}
   }

   \item{
      \textbf{\href{https://fems-microbiology.org/}{Federation of European Microbiological Societies Meeting Attendance Grant}} (2020) \\ 
      for joining the \textit{Metagenomics, Metatranscript- omics and multi 'omics for microbial community studies} Physalia course

   }

   \item{
      \textbf{\href{https://dnaqua.net/stsms/}{Short Term Scientific Mission (STSM) - DNAqua-net COST action}} (2019) \\
      \textbf{Project title:} 
      A comparison of bioinformatic pipelines and sampling techniques to enable benchmarking of DNA metabarcoding \\
      \href{http://dnaqua.net/wp-content/uploads/2019/08/Zafeiropoulos.pdf}{Report}
   }

   \item{
      \textbf{Best Poster Award @ \href{https://hscbio.wordpress.com/}{Hellenic Bioinformatics conference}} (2018) \\
      for \textit{PEMA: a Pipeline for Environmental DNA Metabarcoding Analysis}
   }

\end{itemize}






\section*{Selected presentations}
\begin{itemize}
   \item \textbf{\href{https://www.open-bio.org/events/bosc-2021/}{Bioinformatics Open Source Conference - BOSC2021}} (2021) \\
         \texttt{dingo}: A python library for metabolic networks sampling \& analysis, video poster - \href{https://www.youtube.com/watch?v=IyRD4N6iBu0&t=1s}{video}
   \item \textbf{\href{https://symposium.inrae.fr/dnaqua-conference-evian2021/}{1st DNAQUA International Conference}} (2021)\\
         PEMA v2: addressing metabarcoding bioinformatics analysis challenges, oral talk - \href{https://www.youtube.com/watch?v=kht_LKMmB6w}{video} 
   \item  \textbf{\href{https://fems2020belgrade.com/}{Federation of European Microbiological Societies - FEMS2020}} (2020) \\
         “Mining literature and -omics (meta)data to associate microorganisms, biological processes and environment types” - video poster
   \item \textbf{\href{https://pydata.org/global2020/}{PyData Global PyData2020}} \\
   “Geometric and statistical methods in systems biology: the case of metabolic networks”, oral talk - \href{https://www.youtube.com/watch?v=zg8KFZ_LbHM}{video}
   \item \textbf{\href{http://dnabarcodes2019.org/}{8th International Barcode of Life Conference}} - 2019 \\
   "P.E.M.A.: a pipeline for environmental DNA metabarcoding analysis" (flashtalk)
\end{itemize}


\section*{Participation in proposal writing}

\begin{itemize}
   \item "Climate Change Metagenomic Record Index (CCMRI)" project: submitted at the
   2nd Call for H.F.R.I Research Projects to Support Faculty Members \& Researchers (June 2020). 
   Approved for funding
   \item  "A workflow for marine Genomic Observatories data analysis" project: submitted at the second Training Open Call of EOSC-Life (November 2020). 
   Approved for funding 
\end{itemize}


\section*{Contact}

Personal website: \href{https://hariszaf.github.io/}{https://hariszaf.github.io/} \\
GitHub account: \href{https://github.com/hariszaf}{https://github.com/hariszaf}  \\
Twitter account: \href{https://twitter.com/haris_zaf}{@haris\_zaf} \\
Account in \href{https://www.researchgate.net/profile/Haris-Zafeiropoulos}{ResearchGate} \\
e-mail: \href{mailto:haris.zafr@gmail.com}{haris.zafr@gmail.com}
