\chapter{Metagenome assembled genomes of novel prokaryotic taxa from a hypersaline marsh microbial mat}
\label{app:mags}

\section{MAGs description}

    Using Barrnap~\citep{seemann_barrnap_2014} the 16S rRNA gene was extracted from the retrieved MAGs. 
    For those where a 16S rRNA gene was found, Protologger (version 1.0)~\citep{hitch_automated_2021} 
    was used for a thorough description of their properties. 
    On the Protologger framework, MAG’s closest relatives based on 16S rRNA gene sequence similarity were retrieved using 
    blastn (version 2.12.0+) and The All-Species Living Tree database~\citep{ludwig_release_2021}. 
    Each MAG was placed on the GTDB phylogeny tree using the GTDB-Tk and average nucleotide identity (ANI) values 
    were calculated to check whether the MAG is a representative of an already known species; 
    no ANI value is reported for a genome pair if ANI value is much below 80\%. 

    MAGs were then annotated with Prokka (Seemann 2014b)~\citep{seemann_prokka_2014} 
    and percentage of conserved proteins (POCP) values~\citep{qin_proposed_2014}  
    was calculated between MAGs and the genomes that are close to it based on both the 16S rRNA and the genome-based assignment modules. 
    This was the case only for genomes with validly published names according to the DSMZ nomenclature list. 
    POCP analysis has been used to distinguish prokaryotic genera since a prokaryotic genus can be defined as a group of species 
    with all pairwise POCP values higher than 50\%~\citep{qin_proposed_2014}. 
    The outcome of the Protologger tool for each archaeal MAG can be found on 
    \href{https://github.com/hariszaf/karpathos-swamp/tree/main/MAGs/arc/with_16S}{GitHub} 
    and for each \href{https://github.com/hariszaf/karpathos-swamp/tree/main/MAGs/bac/with_16S}{bacterial} too. 

    For MAG cases suggesting multiple novel entries in novel taxonomic groups higher than the species level, 
    e.g. multiple novel genera within the same family, further POCP values were calculated between each of the MAGs 
    and all of its closest relatives having a genome on GTDB using in-house scripts. 
    For these cases, a phylogenetic tree using the MAG’s alignment by the GTDB-Tk and the genomes’ entries in the 
    GTDB Multiple Sequence Alignment (MSA) was built. 
    Both the phylogenetic trees and the POCP analyses for these cases are available \href{https://github.com/hariszaf/karpathos-swamp/tree/main/MAGs/Phylogenies}{here}. 
    In the “Etymology” section the names given to the new taxa are described. 
    For a thorough investigation of the so-far described characteristics of the reconstructed MAGs’ higher taxonomic levels 
    (e.g., genus, family etc.) the PREGO knowledge base~\citep{zafeiropoulos_prego_2022} was exploited.
    All bioinformatics analyses were supported by the IMBBC High Performance Computing system~\citep{zafeiropoulos_0s_2021}.



\section{Results}

    The vast majority of the taxonomic assignments returned by the GTDB-Tk were 
    at higher than the species level, 
    suggesting that most of the MAGs are representatives of novel taxa; 
    only 4 MAGs were assigned at the species level. 
    For all non low quality MAGs for which the 16S rRNA gene was retrieved.
    Protologger was used to investigate further their uniqueness. 
    Thus, protologues~\citep{tindall_note_1999} accompany 25 archaeal and 100 bacterial 
    MAGs. 
    For the cases where multiple MAGs were supposed to be representatives 
    of a novel higher taxonomic group, e.g. several MAGs suggesting novel genus 
    or genera within a certain family, the extra POCP analyses that were performed 
    along with the corresponding phylogenetic tree are available 
    on GitHub (
        \href{https://github.com/hariszaf/karpathos-swamp/tree/main/MAGs/arc/with_16S}{archaeal} 
        and 
        \href{https://github.com/hariszaf/karpathos-swamp/tree/main/MAGs/bac/with_16S}{bacterial} taxa)
        \footnote{
            Links to the protologues will be publically accessible once the MAGs will 
            be published
        }. 
    In total, our MAGs correspond to the novel taxa present in Table~\ref{table:C1}.


    \begin{table}[h]
        \centering
        \begin{tabular}{lll}
        \hline
        \textbf{Level of novel taxa} & \multicolumn{1}{l|}{\textbf{\# of novel Archaea}} & \multicolumn{1}{l|}{\textbf{\# of novel Bacteria}} \\ \hline
        \textbf{species} & 5 & 11 \\ 
        \textbf{genera} & 13 & 22 \\ 
        \textbf{families} & 4 & 6 \\ 
        \textbf{orders} & 2 & 2 \\
        \textbf{phyla} & - & 1 \\ \cline{1-3}
        \end{tabular}
        \caption{Number of novel taxa described with a protologue based on the MAGs retrieved}
        \label{table:C1}
    \end{table}


    GTDB R07-RS207 was recently released (\href{https://forum.gtdb.ecogenomic.org/t/announcing-gtdb-r07-rs207/264}{2022, April}); 
    the reconstructed MAGs will be 
    classified against this new version of GTDB before publishing 
    to take into account genomes and taxa added in this updated version. 

